src/org/traccar/protocol/Tk103ProtocolDecoder.java

```diff
@@ -190,11 +320,73 @@ private Position decodeNetwork(Channel channel, SocketAddress remoteAddress, Str
 
         getLastLocation(position, null);
 
+        getLastBatteryPower(position);
+
         position.setNetwork(new Network(CellTower.from(
                 parser.nextInt(0), parser.nextInt(0), parser.nextHexInt(0), parser.nextHexInt(0))));
 
         return position;
     }
+
+    private Position decodeLbsWifi(Channel channel, SocketAddress remoteAddress, String sentence) {
+        Parser parser = new Parser(PATTERN_LBSWIFI, sentence);
+        if (!parser.matches()) {
+            return null;
+        }
+
+        DeviceSession deviceSession = getDeviceSession(channel, remoteAddress, parser.next());
+        if (deviceSession == null) {
+            return null;
+        }
+
+        Position position = new Position();
+        position.setProtocol(getProtocolName());
+        position.setDeviceId(deviceSession.getDeviceId());
+
+        decodeType(position, parser.next(), "0");
+
+        getLastLocation(position, null);
+
+        getLastBatteryPower(position);
+
+        Network network = new Network();
+
+        // Parse LBS
```

Comment:
Please remove all unnecessary comments like this.

Reply:
done